src/components/Nav/Nav.tsx

```diff
@@ -60,47 +73,84 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     return this._selectedKey;
   }
 
-  private _renderLink(link: INavLink, linkIndex: number, nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {
+  private _renderAnchorLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
     let { onLinkClick } = this.props;
 
     // Determine the appropriate padding to add before this link.
     // In RTL, the "before" padding will go on the right instead of the left.
     const isRtl: boolean = getRTL();
-    const paddingBefore: string = (_indentationSize * nestingLevel + (hasGroupButton ? 40 : 20)).toString(10) + 'px';
-
-    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
-    if (isLinkSelected) {
-      this._selectedKey = link.key ? link.key : undefined;
-    }
+    const paddingBefore: string = (_indentationSize * nestingLevel +
+      (this._hasExpandButton ? _indentWithExpandButton : _indentNoExpandButton)).toString(10) + 'px';
 
     return (
-      <li role='listitem' key={ linkIndex }>
         <a
-          className={ css('ms-Nav-link', { 'is-selected' : isLinkSelected }) }
+          className={ css('ms-Nav-link', nestingLevel > 0 ? 'ms-Nav-SublinkTextSize' : '') }
           style={ { [isRtl ? 'paddingRight' : 'paddingLeft'] : paddingBefore } }
           href={ link.url || 'javascript:' }
           onClick={ onLinkClick }
           aria-label={ link.ariaLabel }
-          title={ link.title ? link.title : '' }
-          target={ link.target || '' }
+          title={ link.title ? link.title : link.name }
+          target={ link.target }
         >
-          { (link.iconClassName ?
+          { link.iconClassName ?
           <i className={ css('ms-Icon', 'ms-Nav-IconLink', link.iconClassName) }></i>
-          : '') }
+          : '' }
          { this.props.onRenderLink(link)}
         </a>
-        { this._renderLinks(link.links, nestingLevel + 1, hasGroupButton) }
-    </li>
     );
   }
 
-  private _renderLinks(links: INavLink[], nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {
+  private _renderButtonLink(link: INavLink, linkIndex: number) {
+    return (
+      <Button
+          className={ css('ms-Nav-link ms-Nav-linkButton', link.iconClassName) }
+          buttonType={ ButtonType.command }
+          icon='Pencil'
+          description={ link.title || link.name }
+          onClick={ link.onClick }>
+          { link.name }
+      </Button>);
+  }
+
+  private _renderCompositeLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
+    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
+    if (isLinkSelected) {
```

Comment:
Thanks Anton and David,  just integrated & tested the pull request with latest v0.50.0 stuff.  Please take a look, if OK please approve so It can unblock the other components testing that has dependencies on it.

Thanks
-ina

From: David Zearing [mailto:notifications@github.com]
Sent: Wednesday, September 7, 2016 4:46 PM
To: OfficeDev/office-ui-fabric-react <office-ui-fabric-react@noreply.github.com>
Cc: Ina Teegan <inateeg@exchange.microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [OfficeDev/office-ui-fabric-react] Nav component Redline design update (#242)


In src/components/Nav/Nav.tsx<https://github.com/OfficeDev/office-ui-fabric-react/pull/242#discussion_r77925112>:

> -  private _renderLinks(links: INavLink[], nestingLevel: number, hasGroupButton: boolean): React.ReactElement<{}> {

> +  private _renderButtonLink(link: INavLink, linkIndex: number) {

> +    return (

> +      <Button

> +          className={ css('ms-Nav-link ms-Nav-linkButton', link.iconClassName) }

> +          buttonType={ ButtonType.command }

> +          icon='Pencil'

> +          description={ link.title || link.name }

> +          onClick={ link.onClick }>

> +          { link.name }

> +      </Button>);

> +  }

> +

> +  private _renderCompositeLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {

> +    const isLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);

> +    if (isLinkSelected) {

if you split out a separate component like ExpandableLink you could store the isExpanded state within that and not have to have an uber component that tries to do too much.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/OfficeDev/office-ui-fabric-react/pull/242/files/bb5957b9d3278d21be8ae02d015b094d1b5c951a#r77925112>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AT1rLsdSmFJnnRFnBpnSwdXr9e4dIgHAks5qn0zVgaJpZM4J0Kty>.